### PR TITLE
Bug fixes and workarounds

### PR DESCRIPTION
**Pull request by dmannarino**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [x] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

- Creating a raster tile set fails when pixetl computes statistics or histograms
- Failed pixetl jobs unnecessarily run again in Batch
- Creating raster tile caches fails because the batch job name(s) are invalid if a "." is in the version
- Failed batch jobs don't send details back in the changelog
- You can't watch the logs of a batch job in real time (you can only view after finished)
- Uploading PNGs to the tile cache bucket fails due to permissions
- The version of pixetl used fails when running pixetl_prep on an empty prefix
 
Issue Number: GTC-941, GTC-938, GTC-937, GTC-943

## What is the new behavior?

- Creating a raster tile set succeeds unless computing statistics or histograms is explicitly enabled
- Failed pixetl jobs DON'T unnecessarily run again in Batch
- Creating raster tile caches succeeds because batch job name(s) are sanitized
- Failed batch jobs DO send details back in the changelog
- You CAN watch the logs of a batch job in real time
- Uploading PNGs to the tile cache bucket works
- The version of pixetl is increased to one fixing pixetl_prep issue

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: batch/scripts/report_status.sh [hunk 1]
```diff
@@ -12,40 +12,38 @@ URL=${STATUS_URL}/${AWS_BATCH_JOB_ID}
...
+  | sed 's/^PGPASSWORD.*$/PGPASSWORD=\*\*\*/' \
+  | sed 's/^PGUSER.*$/PGUSER=\*\*\*/' \
+  | sed 's/^PGDATABASE.*$/PGDATABASE=\*\*\*/' \
+  | sed 's/^PGHOST.*$/PGHOST=\*\*\*/' \
+  | sed 's/^SERVICE_ACCOUNT_TOKEN.*$/SERVICE_ACCOUNT_TOKEN=\*\*\*/' \
+  | sed 's/^GPG_KEY.*$/GPG_KEY=\*\*\*/' \
+  | tail -c 1000 \
+  > filtered_output.txt
+
+ESC_OUTPUT="$(cat filtered_output.txt | python -c 'import json,sys; print(json.dumps(sys.stdin.read())[1:-1])')"
```
Comment (thomas-maschler) [inline batch/scripts/report_status.sh:37]: Can you try to reuse json_escape() function?

File: batch/scripts/report_status.sh [hunk 2]
```diff
@@ -12,40 +12,38 @@ URL=${STATUS_URL}/${AWS_BATCH_JOB_ID}
 echo "URL: $URL"
 
 # Execute command, save the exit code and output (stdout AND stderr)
-OUTPUT=$("$@" 2>&1)
-EXIT_CODE=$?
+"$@" 2>&1 | tee output.txt
```
Comment (thomas-maschler) [inline batch/scripts/report_status.sh:15]: This is namespaced, right? Somehow I can't spot the section where we create the work directory

**Review by thomas-maschler — APPROVED** [review]

**Comment by thomas-maschler** [comment]
Looks good. Some minor comments, please check if it is worth pursuing.

File: batch/scripts/report_status.sh (hunk 1 above)
Comment (dmannarino) [reply batch/scripts/report_status.sh]: Done!

File: batch/scripts/report_status.sh (hunk 2 above)
Comment (dmannarino) [reply batch/scripts/report_status.sh]: Re-wrote to use a file named after the batch id.

**Comment by thomas-maschler** [comment]
goos job

**Pull request merged**